t/33-vagrant.t

```diff
@@ -0,0 +1,512 @@
+  config.vm.provider "virtualbox" do |v|
+    v.memory = 2042
+    v.cpus = 18
+  end
+end
+END
+    my $virtualbox_backend = backend::vagrant->new();
+    is($virtualbox_backend->{box_name}, $backend_vars->{VAGRANT_BOX},      'Backend variable matches box_name class variable');
+    is($virtualbox_backend->{provider}, $backend_vars->{VAGRANT_PROVIDER}, 'Backend variable matches provider class variable');
+
+    $expected_spurt_str = undef;
+};
+
+subtest 'Vagrantfile with a box url' => sub {
+    $backend_vars->{VAGRANT_PROVIDER} = "libvirt";
+    $backend_vars->{VAGRANT_BOX}      = "Leap-15.2.x86_64";
+    $backend_vars->{VAGRANT_BOX_URL}  = "http://download.opensuse.org/distribution/leap/15.2/appliances/boxes/Leap-15.2.x86_64.json";
+    $backend_vars->{QEMUCPUS}         = 1;
+    $backend_vars->{QEMURAM}          = 1024;
+
+
+    $expected_spurt_str = <<'END';
+Vagrant.configure("2") do |config|
+  config.vm.box = "Leap-15.2.x86_64"
+  config.vm.box_url = "http://download.opensuse.org/distribution/leap/15.2/appliances/boxes/Leap-15.2.x86_64.json"
+  config.vm.synced_folder ".", "/vagrant", disabled: true
+  config.vm.provider :libvirt do |libvirt|
+    libvirt.cpus = 1
+    libvirt.memory = 1024
+    libvirt.storage_pool_name = "vagrant0"
+  end
+end
+END
+    my $provider_with_url = backend::vagrant->new();
+    is($provider_with_url->{box_name}, $backend_vars->{VAGRANT_BOX},      'Backend variable matches box_name class variable');
+    is($provider_with_url->{provider}, $backend_vars->{VAGRANT_PROVIDER}, 'Backend variable matches provider class variable');
+    is($provider_with_url->{box_url},  $backend_vars->{VAGRANT_BOX_URL},  'Backend variable matches box_url class variable');
+
+    $expected_spurt_str = undef;
+    $backend_vars->{VAGRANT_BOX_URL} = undef;
+};
+
+subtest 'backend creation dies when VAGRANT_ASSETDIR cannot be opened' => sub {
+    my $fakedir = '/foo/bar/baz/duitraen/dont/create/this/folder/16';
+    $backend_vars->{VAGRANT_ASSETDIR} = $fakedir;
+    $backend_vars->{VAGRANT_BOX}      = "/foobar";
+
+    eval { backend::vagrant->new(); };
+    like($@, qr/Could not opendir $fakedir/, 'opendir error message is present in the output');
+
+    $backend_vars->{VAGRANT_BOX} = "foobar";
+    $backend_vars->{ASSETDIR}    = undef;
+};
+
+subtest 'backend creation dies when the vagrant box is not in VAGRANT_ASSETDIR' => sub {
+    $backend_vars->{VAGRANT_ASSETDIR} = $tmpdir;
+    $backend_vars->{VAGRANT_BOX}      = "/foobar";
+
+    eval { backend::vagrant->new(); };
+    like(
+        $@, qr/File $tmpdir\/foobar does not exist/,
+        'error message that the box was not found is in the output'
+    );
+
+    $backend_vars->{VAGRANT_BOX} = "foobar";
+    $backend_vars->{ASSETDIR}    = undef;
+};
+
+subtest 'Vagrantfile with a box file existing in VAGRANT_ASSETDIR' => sub {
+    my $box_name = "/Tumbleweed.box";
+    my $box_file = "$tmpdir$box_name";
+    open(FH, ">$box_file");
+    print FH '';
+    close(FH);
```

Comment:
Looks like you're creating an empty file here. Using `Mojo::File::touch` would be nicer.

Reply:
It doesn't look as pretty, since Mojo::file is being mocked, but it's now just one line.

Reply:
Thanks, it is still better than the previous code.